example/spec/ingest/ingestFromPdrSpec.js

```diff
@@ -0,0 +1,441 @@
+'use strict';
+
+/**
+ * End to end ingest from discovering a PDR
+ *
+ * Kick off discover and queue pdrs which:
+ * Discovers 1 PDR
+ * Queues that PDR
+ * Kicks off the ParsePDR workflow
+ *
+ * Parse PDR workflow:
+ * parses pdr
+ * queues a granule
+ * pdr status check
+ * This will kick off the ingest workflow
+ *
+ * Ingest worklow:
+ * runs sync granule - saves file to file staging location
+ * performs the fake processing step - generates CMR metadata
+ * Moves the file to the final location
+ * Does not post to CMR (that is in a separate test)
+ */
+
+const { Collection, Execution, Pdr } = require('@cumulus/api/models');
+const {
+  buildAndExecuteWorkflow,
+  waitForCompletedExecution,
+  LambdaStep,
+  addProviders,
+  cleanupProviders,
+  addCollections,
+  cleanupCollections,
+  api: apiTestUtils
+} = require('@cumulus/integration-tests');
+
+const {
+  loadConfig,
+  updateAndUploadTestDataToBucket,
+  deleteFolder,
+  createTimestampedTestId,
+  createTestDataPath,
+  createTestSuffix,
+  getExecutionUrl
+} = require('../helpers/testUtils');
+
+const {
+  loadFileWithUpdatedGranuleIdPathAndCollection
+} = require('../helpers/granuleUtils');
+
+const config = loadConfig();
+const lambdaStep = new LambdaStep();
+const taskName = 'DiscoverAndQueuePdrs';
+const pdrFilename = 'MOD09GQ_1granule_v3.PDR';
+
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3.PDR',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf'
+];
+
+describe('Ingesting from PDR', () => {
+  const testId = createTimestampedTestId(config.stackName, 'IngestFromPdr');
+  const testSuffix = createTestSuffix(testId);
+  const testDataFolder = createTestDataPath(testId);
+  const providersDir = './data/providers/s3/';
+  const collectionsDir = './data/collections/s3_MOD09GQ_006';
+  const collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+  const provider = { id: `s3_provider${testSuffix}` };
+  process.env.ExecutionsTable = `${config.stackName}-ExecutionsTable`;
+  process.env.CollectionsTable = `${config.stackName}-CollectionsTable`;
+  process.env.PdrsTable = `${config.stackName}-PdrsTable`;
+  const executionModel = new Execution();
+  const collectionModel = new Collection();
+  const pdrModel = new Pdr();
+  let parsePdrExecutionArn;
+
+  beforeAll(async () => {
+    // delete pdr from old tests
+    await apiTestUtils.deletePdr({
+      prefix: config.stackName,
+      pdr: pdrFilename
+    });
+
+    // populate collections, providers and test data
+    await Promise.all([
+      updateAndUploadTestDataToBucket(config.bucket, s3data, testDataFolder, [{ old: 'cumulus-test-data/pdrs', new: testDataFolder }, { old: 'DATA_TYPE = MOD09GQ;', new: `DATA_TYPE = MOD09GQ${testSuffix};` }]),
```

Comment:
This line is pretty long, please break the function call up into separate lines.